cluster-operator/src/main/java/io/strimzi/operator/cluster/model/metrics/MetricsModel.java

```diff
@@ -4,157 +4,22 @@
-     */
-    public String metricsJson(Reconciliation reconciliation, ConfigMap configMap) {
-        if (isEnabled)  {
-            if (configMap == null) {
-                LOGGER.warnCr(reconciliation, "ConfigMap {} does not exist.", configMapName);
-                throw new InvalidConfigurationException("ConfigMap " + configMapName + " does not exist");
-            } else {
-                String data = configMap.getData().get(configMapKey);
-
-                if (data == null) {
-                    LOGGER.warnCr(reconciliation, "ConfigMap {} does not contain specified key {}.", configMapName, configMapKey);
-                    throw new InvalidConfigurationException("ConfigMap " + configMapName + " does not contain specified key " + configMapKey);
-                } else {
-                    if (data.isEmpty()) {
-                        return "{}";
-                    }
-
-                    try {
-                        ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
-                        Object yaml = yamlReader.readValue(data, Object.class);
-                        ObjectMapper jsonWriter = new ObjectMapper();
-
-                        return jsonWriter.writeValueAsString(yaml);
-                    } catch (JsonProcessingException e) {
-                        throw new InvalidConfigurationException("Failed to parse metrics configuration", e);
-                    }
-                }
-            }
-        } else {
-            return null;
-        }
-    }
-
-    /**
-     * Validates the JMX Prometheus Exporter Metrics configuration
-     *
-     * @param jmxMetrics    JMX Prometheus Exporter Metrics configuration which should be validated
-     */
-    /* test */ static void validateJmxPrometheusExporterMetricsConfiguration(JmxPrometheusExporterMetrics jmxMetrics)   {
-        List<String> errors = new ArrayList<>();
-
-        if (jmxMetrics.getValueFrom() != null
-                && jmxMetrics.getValueFrom().getConfigMapKeyRef() != null)   {
-            // The Config Map reference exists
-            if (jmxMetrics.getValueFrom().getConfigMapKeyRef().getName() == null
-                    || jmxMetrics.getValueFrom().getConfigMapKeyRef().getName().isEmpty())  {
-                errors.add("Name of the Config Map with metrics configuration is missing");
-            }
-
-            if (jmxMetrics.getValueFrom().getConfigMapKeyRef().getKey() == null
-                    || jmxMetrics.getValueFrom().getConfigMapKeyRef().getKey().isEmpty())  {
-                errors.add("The key under which the metrics configuration is stored in the ConfigMap is missing");
-            }
-        } else {
-            // The Config Map reference is missing
-            errors.add("Config Map reference is missing");
-        }
-
-        if (!errors.isEmpty())  {
-            throw new InvalidResourceException("Metrics configuration is invalid: " + errors);
-        }
-    }
+    boolean isEnabled();
```

Comment:
When would it exist but not be enabled?

Reply:
It would not. `isEnabled()` now removed from the interface.